contracts/programs/bubblegum/src/lib.rs

```diff
@@ -323,14 +388,51 @@ pub mod bubblegum {
             ctx.accounts.gummyroll_program.to_account_info(),
             gummyroll::cpi::accounts::Initialize {
                 authority: ctx.accounts.authority.to_account_info(),
-                append_authority: ctx.accounts.tree_creator.to_account_info(),
+                append_authority: ctx.accounts.authority.to_account_info(),
                 merkle_roll: merkle_slab,
             },
             authority_pda_signer,
         );
         gummyroll::cpi::init_empty_gummyroll(cpi_ctx, max_depth, max_buffer_size)
     }
 
+    pub fn transfer_tree_owner(
+        ctx: Context<TransferTreeOwner>,
+    ) -> Result<()> {
+        ctx.accounts.authority.owner = ctx.accounts.new_owner.key();
+        Ok(())
+    }
+
+    pub fn transfer_tree_delegate(
+        ctx: Context<TransferTreeDelegate>,
+    ) -> Result<()> {
+        ctx.accounts.authority.delegate = ctx.accounts.new_delegate.key();
+        Ok(())
+    }
+
+    pub fn set_append_authority(
+        ctx: Context<SetTreeAppendAuthority>,
+        index: u8,
+    ) -> Result<()> {
+        if (index as usize) >= ctx.accounts.authority.append_allowlist.len() {
+            return err!(BubblegumError::AppendAllowlistIndexOutOfBounds);
+        }
+        ctx.accounts.authority.append_allowlist[index as usize] = ctx.accounts.new_append_authority.key();
+        Ok(())
+    }
+
+    pub fn remove_append_authority(
+        ctx: Context<RemoveAppendAuthority>,
+    ) -> Result<()> {
```

Comment:
I think the procedure here should be:
1) Locate the index i  of the removed PK
2) Remove it
3) Swap value at i with the last occupied index

Reply:
Basically: create a linked list in memory, remove, then serialize back ?

Reply:
order doesn't matter in the append_list, and this seems like a good way to keep the object packed + prevent duplication

Reply:
I just did a normal `remove`. What's the benefit of doing a `swap remove`?

Reply:
keeps the append list dense

Reply:
I'll come back to this later for cleaner version of remove

Reply:
Added `swap remove`